### PR TITLE
Add new regression tests for DATM-MOM6-CICE6 model

### LINKED ISSUE #224: Add DATM component to s2s
**Description**
The data atmosphere component (DATM) will be added to the s2s repository.

**Solution**
1. The DATM will be added to s2s.
2. Files used by CMake will be added to DATM repository.
3. The DATM-MOM6-CICE6-CMEPS model will be built using CMake.
4. A separate test directory (test_datm) will be created.
5. The parm directory will be moved to test_datm.
6. New baseline results will be created on Hera and Orion.

### PR DESCRIPTION
**Pull request by binli2337**
## Description

1. Create a new directory "tests_datm". It contains files and scripts used for DATM-MOM6-CICE6 model.
2. Move parm directory to tests_datm.
3. Add new regression test files in tests_datm/tests directory.

Are any library updates included in this PR (modulefiles etc.)?
No

### Issue(s) addressed

- fixes #224 

## Testing

How were these changes tested?
Regression tests have been performed on Hera and Orion.

## Dependencies

- waiting on PR #225

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -18,3 +18,25 @@
...
+  branch = develop
+[submodule "CMEPS"]
+  path = CMEPS-interface/CMEPS
+  url = https://github.com/NOAA-EMC/CMEPS.git
+  branch = emc/develop
+[submodule "MOM6"]
+  path = MOM6-interface/MOM6
+  url = https://github.com/NOAA-EMC/MOM6
+  branch = dev/emc
+[submodule "CICE"]
```
Comment (junwang-noaa) [inline .gitmodules]: would you please  merge to the latest ufs-weather develop branch?

**Comment by DeniseWorthen** [comment]
I would suggest updating your 1deg MOM input template to match what is being added on the s2s side. This has the correct settings for the STAGGER=A for both OCEAN_SURFACE_STAGGER and WIND_STAGGER

File: .gitmodules (hunk 1 above)
Comment (binli2337) [reply .gitmodules]: Yes.

**Review by aerorahul — CHANGES_REQUESTED** [review]
@binli2337 
Please merge the branch from #225 in your branch to resolve conflicts.
The `tests_datm/` is not necessary specifically just for data atmosphere.  
Please add the relevant tests and config files for the tests in `tests/`

**Comment by DeniseWorthen** [comment]
There will be conflicts in default_vars between datm and the coupled model.

**Comment by DeniseWorthen** [comment]
You might find it useful to use some of what I put in this [branch](https://github.com/DeniseWorthen/DATM-MOM6-CICE5/tree/feature/cmepscice6). I had been working on it because I needed to be able to have a working branch for the CMEPS changes I want to make for DATM. The 1D RT tests run but I had stopped working on it before getting the 20d tests to run.

It uses the same BL directory layout as the S2S (now weather) for setting up the RTs, the same parm files (where possible) etc. It is set up w/ the 1deg ocean/ice being the default configuration. There are some files in parm that would need to be changed---for example maybe model_configure_datm.IN etc.

**Comment by binli2337** [comment]
This PR is replaced by PR  #256

**Pull request closed without merging**